asset/asset.go

```diff
@@ -488,6 +488,85 @@ func (g *GroupKey) IsEqualGroup(otherGroupKey *GroupKey) bool {
 	return g.GroupPubKey.IsEqual(&otherGroupKey.GroupPubKey)
 }
 
+// IsGroupSig return true if the passed witness is a single Schnorr signature,
+// which is used to perform an asset group reissuance with a signature instead
+// of a script witness.
+func IsGroupSig(witness wire.TxWitness) *schnorr.Signature {
+	var potentialSig []byte
+	if len(witness) == 1 {
+		potentialSig = witness[0]
+	}
+
+	// We can safely drop the signature parsing error, since a witness that
+	// isn't a signature may still be a valid script witness.
+	if len(potentialSig) == schnorr.SignatureSize {
+		groupSig, _ := schnorr.ParseSignature(potentialSig)
+		return groupSig
+	}
+
+	return nil
+}
+
+// ParseGroupWitness parses a group witness that was stored as a TLV stream
+// in the DB.
+func ParseGroupWitness(witness []byte) (wire.TxWitness, error) {
+	var (
+		buf          [8]byte
+		b            = bytes.NewReader(witness)
+		witnessStack wire.TxWitness
+	)
+
+	err := TxWitnessDecoder(b, &witnessStack, &buf, 0)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse group witness: %w", err)
+	}
+
+	return witnessStack, nil
+}
+
+// SerializeGroupWitness serializes a group witness into a TLV stream suitable
+// for storing in the DB.
+func SerializeGroupWitness(witness wire.TxWitness) ([]byte, error) {
+	if len(witness) == 0 {
+		return nil, fmt.Errorf("group witness cannot be empty")
+	}
+
+	var (
+		buf          [8]byte
+		b            bytes.Buffer
+		witnessStack []byte
+	)
+
+	err := TxWitnessEncoder(&b, &witness, &buf)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"unable to serialize group witness: %w", err,
+		)
+	}
+
+	witnessStack = make([]byte, b.Len())
+	copy(witnessStack, b.Bytes())
+	return witnessStack, nil
```

Comment:
I think `b.Bytes()` of a `bytes.Buffer` is safe to use without an additional copy.